### ISSUE TITLE
`_dist` in `Meaning`: either make `tuple` or eliminate

### ISSUE THREAD

**Issue by shanest** [CLMBRs/ultk#31]
`_dist` in `Meaning` is still a dict.   I thought about quickly changing it to a tuple, but wanted to discuss another possibility: just eliminating it all together.   A `Meaning` still has a `Universe`, which itself has a `prior`, so we can get a distribution for the `Meaning` from the `Universe`s.  Are there situations where we want the `Meaning` to be a distribution that comes from some other source than this?

**Comment by nathimel** [comment]
Ah, I tend towards making it a tuple. I had similar thought process when recently working on the recent hashability issues but  for whatever reason forgot to implement.

And yeah, while I think letting a Meaning encode the literal distribution over referents for its expression is clearly an important distinction from the prior ( e.g., color names), we haven't had a use case for this yet since the research ultk supports mainly assumes that every referent is equally, for each Expression/Meaning. But it's nice to have since much of the effcomm research does not make this assumption.

**Comment by shanest** [comment]
Yeah, I totally agree that hard-coding the uniform distribution is the "worst" solution.  So I think the options are (i) `dist` is an arbitrary tuple; if it's not specified in `__init__`, we could default to initialize it according to `self.universe.prior`; or (ii) no `dist` at all in `Meaning`, always just appealing to the `Universe`.  The question is whether (ii) is also too restrictive 🤔

**Comment by nathimel** [comment]
Oh, I see. What does it mean to initialize according to self.universe.prior? To take only the subset of probabilities from the universe prior corresponding to the Meaning.referents, and then renormalize?

**Comment by nathimel** [comment]
This [commit](https://github.com/CLMBRs/ultk/commit/2955a1bfeba0ad7169db8acfb81595b7e94a895d) has the tuple implementation, but defaults to the uniform distribution. 

N.B.: the current modals implementation and indefinites example results assume the uniform distribution even if that is the 'worst', so maybe I'd add a kwarg for uniform initialization to replicate that behavior if we chose not to uniform initialize meanings by default

**Comment by nathimel** [comment]
After further consideration we have decided to explore letting _dist be a frozenset of (Referent, float) tuples. See https://github.com/CLMBRs/ultk/issues/33
